### PR TITLE
Alerting: Use advanced options when query params are not transformable

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes creating a new alert rule UI possible being broken when having query params that are not transformable.

**Why do we need this feature?**

Fixes a bug.

**Who is this feature for?**

All users.
**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx [hunk 1]
```diff
@@ -165,8 +168,12 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   const isGrafanaAlertingType = isGrafanaAlertingRuleByType(type);
   const isRecordingRuleType = isCloudRecordingRuleByType(type);
   const isCloudAlertRuleType = isCloudAlertingRuleByType(type);
+  const queryParamsAreTransformable = areQueriesTransformableToSimpleCondition(dataQueries, expressionQueries);
 
-  const isAdvancedMode = editorSettings?.simplifiedQueryEditor !== true || !isGrafanaAlertingType;
+  const isAdvancedMode =
+    editorSettings?.simplifiedQueryEditor !== true ||
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:174]:
```suggestion
    Boolean(editorSettings?.simplifiedQueryEditor) === false ||
```
nit: equal is a bit easier to read than negation

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx [hunk 2]
```diff
@@ -165,8 +168,12 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   const isGrafanaAlertingType = isGrafanaAlertingRuleByType(type);
   const isRecordingRuleType = isCloudRecordingRuleByType(type);
   const isCloudAlertRuleType = isCloudAlertingRuleByType(type);
+  const queryParamsAreTransformable = areQueriesTransformableToSimpleCondition(dataQueries, expressionQueries);
 
-  const isAdvancedMode = editorSettings?.simplifiedQueryEditor !== true || !isGrafanaAlertingType;
+  const isAdvancedMode =
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:173]: The condition here got pretty complex. Could we add some tests for it?

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:1]: This component got quite complex. I wonder if we could add some tests for it or try to extract some logic to hooks and test these.
I doesn't be part of this PR, but it would be good to keep it in mind

**Review by konrad147 — APPROVED** [review]
LGTM

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:1]: 💯 I take a note for adding them in a separate PR
